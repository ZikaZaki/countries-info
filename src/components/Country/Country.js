import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { HiOutlineArrowCircleRight } from 'react-icons/hi';
import { fetchCountryDetails } from '../../redux/configureStore';

// Import Styling
import styles from './Country.module.css';

const Country = ({
  id,
  //   officialName,
  commonName,
  continent,
  // capital,
  // area,
  // population,
  //   map,
  flag,
}) => {
  const dispatch = useDispatch();

  return (
    <div className={styles['country-card']} key={id}>
      <div className={styles['country-flag']}>
        <img src={flag.png} alt="Country Flag" />
      </div>
      <div className={styles['country-info']}>
        <h2 className={styles['country-name']}>{commonName}</h2>
        <div>
          <p>
            Continent:
            {continent}
          </p>
        </div>
      </div>
      <div className={styles['details-btn']}>
        <Link to="/details" onClick={() => dispatch(fetchCountryDetails(commonName))}>
          <HiOutlineArrowCircleRight />
        </Link>
      </div>
    </div>
  );
};

Country.propTypes = {
  id: PropTypes.string.isRequired,
  //   officialName: PropTypes.string.isRequired,
  commonName: PropTypes.string.isRequired,
  continent: PropTypes.string.isRequired,
  // capital: PropTypes.string.isRequired,
  // area: PropTypes.string.isRequired,
  // population: PropTypes.number.isRequired,
  //   map: PropTypes.string.isRequired,
  flag: PropTypes.string.isRequired,
};

export default Country;
