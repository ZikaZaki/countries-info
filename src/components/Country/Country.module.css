.country-card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  height: 100%;
  width: 100%;
  padding: 0.5rem;
}

.country-card::before {
  mix-blend-mode: difference;
}

.country-card::after {
  mix-blend-mode: difference;
}

.country-map {
  display: flex;
  height: 100%;
  width: 100%;
  padding: 0;
}

.country-map img {
  display: flex;
  width: 100%;
  height: 100%;
}

/* .country-map iframe {
    display: flex;
    width: 100%;
    height: 100%;
    object-fit: contain;
  } */

.country-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  word-wrap: break-word;
  word-break: break-all;
  gap: 0.2rem;
  height: auto;
  width: 100%;
  padding: 0.5rem;
}

.name {
  display: flex;
  font-size: calc(10px + 1vmin);
  font-weight: bold;
  text-align: justify;
  word-wrap: break-word;
  width: 100%;
  padding: 0;
  margin: 0;
}

.country-info p {
  display: flex;
  width: 100%;
  text-align: justify;
  font-weight: 300;
  font-size: 1rem;
  padding: 0;
  margin: 0;
}

.details-btn {
  display: flex;
  font-size: 1.5rem;
  font-weight: bold;
  position: absolute;
  top: 10%;
  right: 15%;
  transform: translate(50%, -50%);
}
