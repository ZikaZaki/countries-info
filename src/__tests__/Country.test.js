import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Country from '../components/Country/Country';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if Item component has changed', () => {
  const flag = {
    png: 'https://restcountries.eu/data/afg.png',
  };
  const country = {
    id: '1',
    commonName: 'Yemen',
    continent: 'Asia',
    png: 'https://restcountries.eu/data/yem.png',
    flag,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Country country={country} flag={flag} />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  const flag = {
    png: 'https://restcountries.eu/data/afg.png',
  };
  const country = {
    id: '1',
    commonName: 'Yemen',
    continent: 'Asia',
    png: 'https://restcountries.eu/data/afg.png',
    flag: {
      png: 'https://restcountries.eu/data/afg.png',
      svg: 'https://restcountries.eu/data/afg.svg',
    },
  };
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Country country={country} flag={flag} />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('country-container');
  expect(container).toBeInTheDocument();
});
