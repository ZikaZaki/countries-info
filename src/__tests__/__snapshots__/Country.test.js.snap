// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Item component has changed 1`] = `
Array [
  <div
    className="country-card"
    data-testid="country-container"
  >
    <div
      className="country-flag"
    >
      <img
        alt="Country Flag"
        src="https://restcountries.eu/data/afg.png"
      />
    </div>
    <div
      className="country-info"
    >
      <h2
        className="country-name"
      />
      <div>
        <p>
          Continent:
        </p>
      </div>
    </div>
    <div
      className="details-btn"
    >
      <a
        href="/details"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </a>
    </div>
  </div>,
  ",",
]
`;
